src/components/views/elements/ImageView.tsx

```diff
@@ -160,41 +160,82 @@ export default class ImageView extends React.Component<IProps, IState> {
 
         if (newZoom <= this.state.minZoom) {
+            // Zoom out fully
             this.setState({
                 zoom: this.state.minZoom,
                 translationX: 0,
                 translationY: 0,
             });
-            return;
-        }
-        if (newZoom >= this.state.maxZoom) {
-            this.setState({ zoom: this.state.maxZoom });
-            return;
-        }
+        } else if (zoomX === undefined && zoomY === undefined) {
+            // Zoom relative to the center of the view
+            this.setState({
+                zoom: newZoom,
+                translationX: this.state.translationX * newZoom / oldZoom,
+                translationY: this.state.translationY * newZoom / oldZoom,
+            });
+        } else {
+            // Zoom relative to the given point on the image.
+            // First we need to figure out the offset of the anchor point
+            // relative to the center of the image, accounting for rotation.
+            let offsetX;
+            let offsetY;
+            switch (((this.state.rotation % 360) + 360) % 360) {
+                case 0:
+                    offsetX = this.image.current.clientWidth / 2 - zoomX;
+                    offsetY = this.image.current.clientHeight / 2 - zoomY;
+                    break;
+                case 90:
+                    offsetX = zoomY - this.image.current.clientHeight / 2;
+                    offsetY = this.image.current.clientWidth / 2 - zoomX;
+                    break;
+                case 180:
+                    offsetX = zoomX - this.image.current.clientWidth / 2;
+                    offsetY = zoomY - this.image.current.clientHeight / 2;
+                    break;
+                case 270:
+                    offsetX = this.image.current.clientHeight / 2 - zoomY;
+                    offsetY = zoomX - this.image.current.clientWidth / 2;
+            }
 
-        this.setState({
-            zoom: newZoom,
-        });
+            // Apply the zoom and offset
+            this.setState({
+                zoom: newZoom,
+                translationX: this.state.translationX + (newZoom - oldZoom) * offsetX,
+                translationY: this.state.translationY + (newZoom - oldZoom) * offsetY,
+            });
+        }
     }
 
     private onWheel = (ev: WheelEvent) => {
         ev.stopPropagation();
         ev.preventDefault();
-
         const { deltaY } = normalizeWheelEvent(ev);
-        this.zoom(-(deltaY * ZOOM_COEFFICIENT));
+
+        if (ev.target === this.image.current) {
+            // Zoom in on the point on the image targeted by the cursor
+            this.zoomDelta(-deltaY * ZOOM_COEFFICIENT, ev.offsetX, ev.offsetY);
+        } else {
+            // The user is scrolling outside of the image, so we can't really
+            // get a targeted point. Instead, we'll just zoom in on the center.
+            this.zoomDelta(-deltaY * ZOOM_COEFFICIENT);
```

Comment:
I think we actually shouldn't do anything here

Reply:
Maybe, was it intentional that you can zoom in on images without hovering directly over them, or was that just because we were too lazy to have the additional check?

Reply:
TBH, I didn't even think of this, so it was very much unintentional :sweat_smile: